File: src/main/java/com/back/global/rateLimit/RateLimitPolicy.java
package com.back.global.rateLimit;

import lombok.AllArgsConstructor;
import lombok.Getter;

import java.time.Duration;

@Getter
@AllArgsConstructor
public enum RateLimitPolicy {

    LOGIN(5, Duration.ofMinutes(1)),
    API_DEFAULT(100, Duration.ofMinutes(1)),
    API_HEAVY(20, Duration.ofMinutes(1));

    private final int limit;
    private final Duration duration;
}

File: src/main/java/com/back/global/rateLimit/RateLimitPolicy.kt
package com.back.global.rateLimit

import java.time.Duration

enum class RateLimitPolicy(
    val limit: Int,
    val duration: Duration
) {
    LOGIN(500000000, Duration.ofMinutes(1)),
    API_DEFAULT(10000000, Duration.ofMinutes(1)),
    API_HEAVY(2000000000, Duration.ofMinutes(1))
}